python/ctsm/subset_data.py

```diff
@@ -0,0 +1,893 @@
+            lon2,
+            reg_name,
+            create_domain,
+            create_surfdata,
+            create_landuse,
+            create_datm,
+        )
+
+        print(region)
+
+        if crop_flag:
+            num_pft = "78"
+        else:
+            num_pft = "16"
+
+        print(" crop_flag = " + crop_flag.__str__() + " num_pft =" + num_pft)
+
+        region.create_tag()
+
+        # --  Set input and output filenames
+        # --  Specify input and output directories
+        dir_output = "/glade/scratch/" + myname + "/region/"
+        if not os.path.isdir(dir_output):
+            os.mkdir(dir_output)
+
+        dir_inputdata = "/glade/p/cesmdata/cseg/inputdata/"
+        dir_clm_forcedata = "/glade/p/cgd/tss/CTSM_datm_forcing_data/"
+
+        # --  Set time stamp
+        command = 'date "+%y%m%d"'
+        x2 = subprocess.Popen(command, stdout=subprocess.PIPE, shell="True")
+        x = x2.communicate()
+        timetag = x[0].strip()
+        print(timetag)
+
+        # --  Specify land domain file  ---------------------------------
+        fdomain_in = (
+            dir_inputdata + "share/domains/domain.lnd.fv1.9x2.5_gx1v7.170518.nc"
+        )
+        fdomain_out = (
+            dir_output + "domain.lnd.fv1.9x2.5_gx1v7." + region.tag + "_170518.nc"
+        )
+        # SinglePointCase.set_fdomain (fdomain)
+        region.fdomain_in = fdomain_in
+        region.fdomain_out = fdomain_out
+
+        # --  Specify surface data file  --------------------------------
+        fsurf_in = (
+            dir_inputdata
+            + "lnd/clm2/surfdata_map/surfdata_1.9x2.5_78pfts_CMIP6_simyr1850_c170824.nc"
+        )
+        fsurf_out = (
+            dir_output
+            + "surfdata_1.9x2.5_78pfts_CMIP6_simyr1850_"
+            + region.tag
+            + "_c170824.nc"
+        )
+        region.fsurf_in = fsurf_in
+        region.fsurf_out = fsurf_out
+
+        # --  Specify landuse file  -------------------------------------
+        fluse_in = (
+            dir_inputdata
+            + "lnd/clm2/surfdata_map/landuse.timeseries_1.9x2.5_hist_78pfts_CMIP6_simyr1850-2015_c170824.nc"
+        )
+        fluse_out = (
+            dir_output
+            + "landuse.timeseries_1.9x2.5_hist_78pfts_CMIP6_simyr1850-2015_"
+            + region.tag
+            + ".c170824.nc"
+        )
+        region.fluse_in = fluse_in
+        region.fluse_out = fluse_out
+
+        # --  Create CTSM domain file
+        if create_domain:
+            region.create_domain_at_reg()
+
+        # --  Create CTSM surface data file
+        if create_surfdata:
+            region.create_surfdata_at_reg()
+
+        # --  Create CTSM transient landuse data file
+        if create_landuse:
+            region.create_landuse_at_reg()
+        print("Successfully ran script for a regional case.")
+
+    else:
+        # print help when no option is chosen
+        get_parser().print_help()
+
+
+if __name__ == "__main__":
+    main()
```

Comment:
- [ ] As mentioned above, I'd get rid of this.

Reply:
- [ ] Also: change file to not be executable (with chmod) and get rid of the shebang line in this file.